test/TestUtilities/Test.Utility/Signing/TestCertificate.cs

```diff
@@ -36,18 +37,31 @@ public class TestCertificate
         /// Trust the PublicCert cert for the life of the object.
         /// </summary>
         /// <remarks>Dispose of the object returned!</remarks>
-        public TrustedTestCert<TestCertificate> WithTrust(StoreName storeName = StoreName.TrustedPeople, StoreLocation storeLocation = StoreLocation.CurrentUser)
+        /// According to https://github.com/dotnet/corefx/blob/master/Documentation/architecture/cross-platform-cryptography.md#x509store
+        /// Linux could not read/write LocalMachine\Root , but could only read/write CurrentUser\Root
+        public TrustedTestCert<TestCertificate> WithTrust()
         {
-            return new TrustedTestCert<TestCertificate>(this, e => PublicCert, storeName, storeLocation);
+
+            if (RuntimeEnvironmentHelper.IsLinux)
+            {
+                return new TrustedTestCert<TestCertificate>(this, e => PublicCert, StoreName.Root, StoreLocation.CurrentUser);
+            }
+           
+            return new TrustedTestCert<TestCertificate>(this, e => PublicCert, StoreName.Root, StoreLocation.LocalMachine);
```

Comment:
shouldn't it be CurrentUser for OSX too?

Reply:
Thanks! Pls refer to https://github.com/NuGet/NuGet.Client/pull/3168#discussion_r362672651